### PR TITLE
[WIP] Refactor

### PR DESCRIPTION
**Pull request by izaid**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -224,33 +224,33 @@ set(pydynd_CPP_SRC
     )
 set(pydynd_CYTHON_SRC
     dynd/cuda.pyx
-    dynd/dynd._pydynd.pyx
-    dynd/array.pxd
+    dynd/_pydynd.pyx
```
Comment (mwiebe) [inline CMakeLists.txt]: The dynd._pydynd name was to make it possible to pickle the types, see https://github.com/libdynd/dynd-python/issues/96. Did you figure out an alternative way to allow this?

**Comment by izaid** [comment]
Ah, okay, there was no test for this. I'll investigate this when I get a chance.

**Comment by mwiebe** [comment]
My bad, I clearly should have tested that originally.

**Comment by insertinterestingnamehere** [comment]
I remember seeing some similar work on serializing kd trees in SciPy and scikit-learn. Supposedly the [SciPy version](https://github.com/scipy/scipy/pull/4374#issuecomment-69269226) only works for Python 3 and the [scikit-learn version](https://github.com/scikit-learn/scikit-learn/blob/2d559ba20a7102aee140ef7d2f49bf279b347b06/sklearn/neighbors/binary_tree.pxi#L1097) works for both.

**Comment by mwiebe** [comment]
@insertinterestingnamehere I think those two references are for pickling the object values, not the object type as is the case here. Am a reading that correctly?

**Comment by insertinterestingnamehere** [comment]
@mwiebe Yes, I misunderstood the original issue. Thanks for pointing that out. I don't know if the other examples are still relevant.

**Pull request merged**